k2/csrc/ragged_ops.h

```diff
@@ -347,6 +347,70 @@ inline Ragged<T> RaggedFromTotSizes(ContextPtr &c,
                    Array1<T>(c, tot_sizes.back()));
 }
 
+/*
+  Transpose a ragged tensor as if it were the index information of a CSR-format
+  sparse matrix (but with possibly repeated elements!).  This is easiest to
+  explain if we assume `src` has 2 axes.  We view `src` as a list of nonzero
+  elements of a matrix, indexed first by row, and containing column-indexes
+  (but possibly repeated column indexes, which violates the assumptions of
+  the cusparse library).  This function returns an array of dimension
```

Comment:
@danpovey 
I would like to give an example to show my understanding of the comment.
Please correct me if I am wrong.

For the following CSR-format matrix:
```
0 0 0 0
5 8 0 0
0 0 3 0
0 6 0 0
```


- row splits: 0 0 2 3 4
- values:       5 8 3 6

Transpose of the matrix is:

```
0 5 0 0
0 8 0 6
0 0 3 0
0 0 0 0
```
The values are `5 8 6 3`

So the returned `Array<int32_t>` contains: 0 1 3 2, which
maps `5 8 3 6` to `5 8 6 3`.

Reply:
If the above example is correct, I would like to implement it
by using the transpose algorithm from this paper:
[Two Fast Algorithms for Sparse Matrices: Multiplication and Permuted Transposition][1]

[gustavson1978.pdf](https://github.com/k2-fsa/k2/files/5355804/gustavson1978.pdf)


[1]: https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=Two+Fast+Algorithms+for+Sparse+Matrices%3A+Multiplication+and+Permuted+Transposition&btnG=

Reply:
why is `values` here so large? I was thinking they are column indexes (so any value should be `<4`)

Reply:
> The values are 5 8 6 3

This is for readability.

---

The following is how it looks like in code

https://github.com/k2-fsa/k2/blob/a531d3e0b2f6ecda1fd1ca45227289613ca4fa09/k2/csrc/ragged_test.cu#L1020-L1025

Reply:
aha, OK.